packages/flutter_tools/templates/app_shared/windows.tmpl/runner/win32_window.cpp

```diff
@@ -126,6 +131,9 @@ bool Win32Window::Create(const std::wstring& title,
     return false;
   }
 
+  BOOL honorDarkMode = TRUE;
+  DwmSetWindowAttribute(window, DWMWA_USE_IMMERSIVE_DARK_MODE, &honorDarkMode, sizeof(honorDarkMode));
```

Comment:
Does setting this window attribute allow the Flutter app context to use the specified dark mode theme instead of its default if provided, or does it affect only the part of the window outside of the Flutter app content?

Reply:
Good questions! I added more details to the pull request's description to clarify.

> does it affect only the part of the window outside of the Flutter app content?

Yup, that's correct.

> Does setting this window attribute allow the Flutter app context to use the specified dark mode theme instead

This will be done in a future change. We'll need to respond to Windows settings change message in the top-level message handler.